packages/framework/agent-scheduler/src/scheduler.ts

```diff
@@ -181,8 +181,20 @@ export class AgentScheduler
 					taskUrl: { tag: TelemetryDataTag.CodeArtifact, value: taskUrl },
 				});
 			}
-			// Note - the assumption is - we are connected.
-			// If not - all tasks should have been dropped already on disconnect / attachment
+			if (!this.runningTasks.has(taskUrl)) {
+				// If we got disconnected (and are attached), tasks that we WERE picked for at the time of disconnect
+				// will still show us as holding the task according to getTaskClientId (the CRC is stale), but we
+				// should not try to release because our disconnect will already result in either someone else or
+				// ourselves clearing the task upon reconnect.
+				// This UsageError is to enforce that the caller should check AgentScheduler.pickedTasks before trying
```

Comment:
> What use cases lead us here - did the client pick the task, then got disconnected, or they never pick the task to start with (or both)?

Either one can lead here - both result in the client NOT having the task picked.  Basically:

Recommendation: Always check that the task is picked before trying to release() it.  If you try to release an unpicked task:

Today: Sometimes you'll blow up with a usage error (if you're connected), sometimes you'll blow up with an assert (if you're disconnected).  But you'll always blow up.

With this PR: You'll always blow up with a usage error.

Reply:
Thanks for the clarification here, @ChumpChief. Given the change has a relatively small impact, I'm ok with moving this in. However, few of the questions above still stand. Most importantly: Where do we guide users around these recommendations? If we are planning to support this package going forward, at the very least we should have content in place that allows anyone (reviews/consumers) to reason about the overall API usage and correctness. If that's a bigger lift, it can be separate work item / PR.

Reply:
Let’s chat about this then. Unless we have some clear/important gap it’s filling now, we should be able to discourage the use without necessarily waiting on TaskManager. At least it would make it clear why we are not maintaining the package and documenting APIs properly.